### PR TITLE
JMicrodata update

### PR DESCRIPTION
**Pull request by alexprut**
— Updated the types.json file with the new available Types and Properties
— Small documentation & code refactoring
— Bug fix: if specified, return an empty string on fallback or disabled library, even if a content is available https://github.com/joomla/joomla-cms/pull/4108/files#diff-cd3328236e7aecb09373c1c564b9bbb8R256

### CODE REVIEW

**Comment by roland-d** [comment]
Hello @PAlexcom

Thank you for your contribution.

The last comment here was on 16th August 2014. So the question is, Is this issue/pull request still valid?
if so please provide clear test instructions to be able to test / reproduce this issue.
If no reply is received within 4 weeks we will close this issue.

Thanks for understanding! <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4108">issues.joomla.org/joomla-cms/4108</a>.</sub>

**Comment by PAlexcom** [comment]
Hi @roland-d, this PR is still valid, I can't provide test instructions to test / reproduce this issue simply because this PR contains some code refactoring/bug fix (with tests)/update of the JMicrodata library, the PR can be merged.

File: libraries/joomla/microdata/microdata.php [hunk 1]
```diff
@@ -773,113 +771,85 @@ public static function isTypeAvailable($type)
...
 	 * @param   string   $scope     Optional, the Type scope to display
-	 * @param   boolean  $inverse   Optional, default = false, inverse the $scope with the $property
+	 * @param   boolean  $invert    Optional, default = false, invert the $scope with the $property
 	 *
 	 * @return  string
 	 *
-	 * @since	3.2
+	 * @since   3.2
 	 */
-	public static function htmlMeta($content, $property, $scope = '', $inverse = false)
```
Comment (roland-d) [inline libraries/joomla/microdata/microdata.php]: Can you add the deprecated notice?

**Comment by roland-d** [comment]
Hello @PAlexcom the bug fix should be reproducible or not? I see that some kind off fallback mechanism has changed, shouldn't that be testable? The code is touched somewhere or not?

**Comment by roland-d** [comment]
@PAlexcom Can you provide feedback on my questions?

**Comment by roland-d** [comment]
Merged based on review. Library is not used in core itself.

**Comment by peterpeter** [comment]
@PAlexcom Nice and usefull piece of code, the JMicrodata class. Thanks.
I started these days to implement microdata in some extensions, but there are differences between schema.org and the types.json file.

I found some missing properties (MusicEvent->workPerformed) and expected porperty-types (PostalAddress->addressCountry can be Text too, not only type 'Country')

To avoid hardcoding - where can I get the most recent types.json-file? Should I use your Spider4Schema for that?

**Comment by alexprut** [comment]
@roland-d I really apologize for not replying in time, I see now that the PR was merged, it should not break anything and update the new schema, to be honest two years have passed: I do not remember exactly what bug the PR fixes, but I can investigate if needed.

**Comment by alexprut** [comment]
@peterpeter Thanks & sorry for the late reply ;)
> I started these days to implement microdata in some extensions, but there are differences between schema.org and the types.json file.

That is because the `types.json` file is an older version, a new PR should be done in order to update the file.

> To avoid hardcoding - where can I get the most recent types.json-file? Should I use your Spider4Schema for that?

Yes, Spider4Schema is the spider/parser that creates the `types.json` file, however I'm not sure if it still works properly since schema.org have changed their website recently.

**Comment by roland-d** [comment]
@alexprut I understand you don't remember when it's such an old PR. It is already in Joomla 3.5.0 so it's OK. At least it did go in :)

I hardly dare to ask but could you do a new PR with the updates types.json file? Put me in there as well so I am aware of it. Thank you.

**Comment by alexprut** [comment]
@roland-d I'll try do my best to update the file asap.

**Comment by roland-d** [comment]
@alexprut Thanks, we are here all weekend working on the issue tracker, if you can get it done we will merge it this weekend for sure.

**Pull request merged**